cmake/Modules/Options.cmake

```
@@ -25,26 +27,26 @@ endif()
 # Check if CUDA can be enabled
 find_package(CUDA)
 if(CUDA_FOUND)
-    set(_USE_CUDA ON)
+    check_language(CUDA)
+    if(CMAKE_CUDA_COMPILER)
+        enable_language(CUDA)
+    else()
+        set(_USE_CUDA OFF)
+    endif()
```

```suggestion
    set(_USE_CUDA ON)
```
We cannot enable CUDA here because we haven't yet checked whether the user has asked for CUDA. `enable_cuda` will stop the build if it decided the compiler is not working correctly.